pkg/gardenlet/controller/backupbucket/backup_bucket_reconciler.go

```diff
@@ -97,6 +97,7 @@ func (r *reconciler) reconcileBackupBucket(ctx context.Context, gardenClient kub
 			backupBucketLogger.Errorf("Failed to ensure gardener finalizer on backupbucket: %+v", err)
 			return reconcile.Result{}, err
 		}
+		return reconcile.Result{}, nil
```

Comment:
BackupBucket reconciliation has the same "issue" as the project control, where the `backupBucket.Status.ObservedGeneration` is set to equal the `backupBucket.Generation` at the end of the reconcilation.